File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int i = 0;
        int left = 0;
        int right = nums.size() - 1;
        while(i <= right){
            if (nums[i] == 0){
                int temp = nums[i];
                nums[i] = nums[left];
                nums[left] = temp;
                i++;
                left++;  
            }
            else if (nums[i] == 2){
                int temp = nums[i];
                nums[i] = nums[right];
                nums[right] = temp;
                right--;
            }
            else{
                i++;
            }
        } 
    }
};

File: 0075-sort-colors/0075-sort-colors.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        if nums.is_empty() {  
            return;
        }
        
        let mut i: usize = 0;
        let mut left: usize = 0;
        let mut right: usize = nums.len() - 1;
        
        while i <= right {
            if nums[i] == 0 {
                nums.swap(i, left);
                i += 1;
                left += 1;
            } else if nums[i] == 2 {
                nums.swap(i, right);
                if right == 0 { 
                    break;
                }
                right -= 1;
            } else {
                i += 1;
            }
        }
    }
}